### ISSUE TITLE
2x2 Rotor, or Stretchable?

### ISSUE THREAD

**Issue by Vicomt** [tsunamayo/Starship-EVO#2000]
### Short Description of the issue:

please for the love of all that's Holy and French, can we have a 2x2 block rotator, or the ability to stretch the rotor block?
...

### Steps to reproduce the issue:
attempt to place a rotation origin on the edge of a block.

**Comment by Garrett-C** [comment]
I too think this could be a good addition. I was recently building a large turret (Not talking monster turret but certainly big) but the turret has surpassed the mass limit on rotors, a way to resize the rotors like the hinges and rails would be a good addition. 

Here is an image of the turret for reference
![image](https://user-images.githubusercontent.com/43644633/81424810-6cf9ac80-914e-11ea-91a6-4557d7128774.png)
![image](https://user-images.githubusercontent.com/43644633/81424863-813da980-914e-11ea-96ea-0f982f4caaa2.png)

**Comment by ProPeach** [comment]
I think there are two different points being put across here, the aesthetic argument and the functionality argument.
On the one hand 2x2 rotors would be cool for even numbered builds and for adding a rotational axis on the corner of the block.
On the other, the current rotor has a very small mass limit so we can't build large functional turrets. Larger turrets might be better for performance as they do the job of multiple smaller turrets, and they're important for building and combat styles too.

I think it's important that a stretchable rotor only increases its mass limit by stretching its height. If a 2x2 rotor has a larger mass limit than a 1x1 rotor, players who want to build larger turrets will find themselves forced to build using an even numbered grid which is will cause problems, as the vast majority of ships use an odd width.

Here's what I'm talking about - Rotors of the same width should have the same mass limit, but you can increase it by adding to their height
![image](https://user-images.githubusercontent.com/27857711/81435427-c4544880-915f-11ea-993a-317538ad2610.png)

**Comment by Sullos** [comment]
I agree that the adjustable rotor would be a great addition. I think everyone starts to notice that the brick rotor is too small for what feels right with the brick-sized weapons, but the block-sized rotor is too big for brick weapons. I think  2x2 is the minimal option of what we could very much appreciate. But Peaches idea really works well :)

**Comment by captain828** [comment]
> Here's what I'm talking about - Rotors of the same width should have the same mass limit, but you can increase it by adding to their height

I think it should work exactly how it does for the stretchable thrusters and weapons: it keeps a counter of how many normal variants would fit in the volume (shown as "units" in the UI).

So basically, a single rotor unit would support 10t, with it scaling linearly after that: a 4 unit motor should support 40t etc.
This would also mean that the current large rotor block should be updated with the new limits, so instead of 50t, as it is currently, it should support more like 160t (I think you can fit 16 small blocks inside a large one?).

Similarly, rails should be updated to support this scheme.

**Comment by tsunamayo** [comment]
I wonder how they do that in other game, where there is no stretching? ie starmade / SE ect...
A vert stretch could be easy to do, but the flat stretch a bit less I need to be careful with this I dont want to break that before ea. (but I dont want to break it after too ;) )
Cheers

**Comment by Garrett-C** [comment]
> I wonder how they do that in other game, where there is no stretching? ie starmade / SE ect...
> A vert stretch could be easy to do, but the flat stretch a bit less I need to be careful with this I dont want to break that before ea. (but I dont want to break it after too ;) )
> Cheers

In terms of other games I belive starmade has some form of addon blocks than can be placed and connected to add to the mass. I don't believe they have any system to allow you to have a 2x2 rotor. They actually have rotors and rails as the same system, which may be worth investigating at some stage to allow for more elaborate rails. 

But yeah I don't think any game has a system to allow a 2x2 esc rotor to be placed. I could be wrong though.

**Comment by SageThe13th** [comment]
From The Depths has 1x1, 3x3, and 5x5 rotating blocks. The difference is in health since these blocks don't have a weight limit.

**Comment by cloudrambler** [comment]
@tsunamayo SE has a torque limit per rotator, which effectively determines how much mass can be moved before you risk overloading the rotor (things explode in SE if they get overloaded, and that's not part of the game design, it just happens when the physics engine freaks out).

SE also allows actuators to be connected in parallel so that their effects are accumulated. This is done for rotors by placing them on a common axis, able to be fixed to both bodies (parent and child). This is achieved by placing the rotor head as a separate block to the rotor body and then merging them into a single operable device. This method allows complex mechanisms to be made with closed loops of driven and passive actuators.

IMO @ProPeach suggestion is spot on and would be simpler than SE for basic movement arrangements.

**Comment by tsunamayo** [comment]
Now we have grid-scale for rotators.
